pallets/pallet-deposit-storage/src/fungible/tests/transfer.rs

```diff
@@ -0,0 +1,102 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2024 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use frame_support::traits::{
+	fungible::MutateHold,
+	tokens::{Fortitude, Precision, Preservation, Restriction},
+};
+use kilt_support::Deposit;
+use sp_runtime::AccountId32;
+
+use crate::{
+	deposit::DepositEntry,
+	fungible::{
+		tests::mock::{ExtBuilder, TestRuntime, TestRuntimeHoldReason, OTHER_ACCOUNT, OWNER},
+		PalletDepositStorageReason,
+	},
+	Pallet, SystemDeposits,
+};
+
+#[test]
+fn transfer_on_hold() {
+	ExtBuilder::default()
+		.with_balances(vec![(OWNER, 100_000), (OTHER_ACCOUNT, 1)])
+		.build()
+		.execute_with(|| {
+			let reason = PalletDepositStorageReason::default();
+
+			<Pallet<TestRuntime> as MutateHold<AccountId32>>::hold(&reason, &OWNER, 10)
+				.expect("Failed to hold amount for user.");
+
+			<Pallet<TestRuntime> as MutateHold<AccountId32>>::transfer_on_hold(
```

Comment:
What is the expected behavior of transfer_on_hold?

Reply:
From the docs:

```
        /// Transfer held funds into a destination account.
	///
	/// If `mode` is `OnHold`, then the destination account must already exist and the assets
	/// transferred will still be on hold in the destination account. If not, then the destination
	/// account need not already exist, but must be creatable.
	///
	/// If `precision` is `BestEffort`, then an amount less than `amount` may be transferred without
	/// error.
	///
	/// If `force` is `Force`, then other fund-locking mechanisms may be disregarded. It should be
	/// left as `Polite` in most circumstances, but when you want the same power as a `slash`, it
	/// may be `Force`.
	///
	/// The actual amount transferred is returned, or `Err` in the case of error and nothing is
	/// changed.
```

Reply:
To be fair, I did not extensively test all the `MutateHold` functions. It is mostly like a smoke test to make sure we don't deviate from the "traditional" implementation.